examples/tv-casting-app/android/App/app/src/main/java/com/chip/casting/app/CertTestFragment.java

```diff
@@ -262,6 +304,45 @@ private void runCertTests(Activity activity) {
               kContentApp, successFailureCallbackInteger, successFailureCallbackInteger);
         });
 
+    runAndWait(
+        "mediaPlayback_subscribeToCurrentState",
+        successFailureCallback,
+        () -> {
+          tvCastingApp.mediaPlayback_subscribeToCurrentState(
+              kContentApp,
+              new SuccessCallback<MediaPlaybackTypes.PlaybackStateEnum>() {
+                @Override
+                public void handle(MediaPlaybackTypes.PlaybackStateEnum response) {
+                  // Lets wait for the timeout to avoid the race condition issue in the SDK with
```

Comment:
I do not understand the comment: there is no wait here (so why `lets wait`?) and what is this in relation with ReadClient::Close and successFailureCallback.handle?

Reply:
When the test starts it creates a countdown latch with count of 1. The main thread then waits on the latch to down to 0. Actual call is made async. In the handle call it decrements the latch to allow the main thread to start the next test. Hence is handle is not call, the main thread will time out(5 secs).
I will create an issue for the ReadClient race condition. Waiting for this PR to be merged so that I can include this test as a way to reproduce. Problem is that if the ReadClient is processing an incoming message and at the same  time the SDK gets a call to shutdown subscription for that same ReadClient, there is a race condition where by the code uses an invalid reference and leads to a crash.